src/wallet/wallet.cpp

```diff
@@ -3913,6 +3913,28 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
     // interface.
     walletInstance->m_chain_notifications_handler = walletInstance->chain().handleNotifications(walletInstance);
 
+    // Unless allowed, ensure wallet files are not reused across chains:
+    if (!gArgs.GetBoolArg("-walletcrosschain", DEFAULT_WALLETCROSSCHAIN))
+    {
+        WalletBatch batch(*walletInstance->database);
+        CBlockLocator locator;
+        if (batch.ReadBestBlock(locator))
+        {
+            // Wallet is assumed to be from another chain, if none of
+            // its last 6 best known blocks are in the active chain
+            // (this heavily relies on the fact that locator stores
+            // last 10 blocks consecutively):
+            const Optional<int> fork_height = chain.findLocatorFork(locator);
```

Comment:
Please [use `std::optional` in new code](https://github.com/bitcoin/bitcoin/blob/master/src/optional.h#L12).

Reply:
done